### PR TITLE
Add autonomous gear scoring

### PR DESCRIPTION
**Pull request by dengzac**

### CODE REVIEW

File: src/Commands/EjectGear.h [hunk 1]
```diff
@@ -1,17 +1,19 @@
-#ifndef AUTO_COMMAND_H
-#define AUTO_COMMAND_H
+#ifndef EJECT_GEAR_H
```
Comment (colescott) [inline src/Commands/EjectGear.h:1]: Git 3 gud 5 me

File: src/Subsystems/Drivetrain.h [hunk 2]
```diff
@@ -7,16 +7,18 @@
...
+#include "AHRS.h"
 
 class Drivetrain : public Subsystem
 {
     public:
         Drivetrain();
         void InitDefaultCommand();
         void Set ( double left, double right );
-        void SetRaw ( double left, double right );
+        void SetRaw ( double left, double right, bool currentLimit = true );
```
Comment (colescott) [inline src/Subsystems/Drivetrain.h:21]: ;-; y u no like my current limit?

File: src/Subsystems/Drivetrain.h [hunk 3]
```diff
@@ -25,6 +27,9 @@ class Drivetrain : public Subsystem
         void EnablePID();
         void DisablePID();
 
+		double GetLeftPIDOutput();
+		double GetRightPIDOutput();
+		bool DistancePIDDone();
         struct DrivetrainVoltage {
```
Comment (colescott) [inline src/Subsystems/Drivetrain.h]: this stuff can go with this PR. delete all voltage stuff.

File: src/Subsystems/Drivetrain.h [hunk 4]
```diff
@@ -53,15 +67,23 @@ class Drivetrain : public Subsystem
...
+        const double PID_LEFT_F = 0.0;
+		const double PID_LEFT_TOLERANCE = 2;
 
-        const double PID_RIGHT_P = 0.15;
+        const double PID_RIGHT_P = 0.1;
         const double PID_RIGHT_I = 0.0;
         const double PID_RIGHT_D = 0.01;
-        const double PID_RIGHT_F = 0.05;
+        const double PID_RIGHT_F = 0;
+		const double PID_RIGHT_TOLERANCE = 2;
```
Comment (colescott) [inline src/Subsystems/Drivetrain.h:64]: 2 inches? it could be bigger than that.

File: src/Subsystems/Drivetrain.h [hunk 5]
```diff
@@ -84,7 +106,13 @@ class Drivetrain : public Subsystem
...
         std::unique_ptr<frc::PIDController> LeftPID;
         std::unique_ptr<frc::PIDController> RightPID;
+		std::unique_ptr<PIDOutputReceiver> LeftOutput;
+		std::unique_ptr<PIDOutputReceiver> RightOutput;
 
+		std::unique_ptr<frc::PIDController> TurnPID;
+		std::unique_ptr<AHRS> ahrs;
+		std::unique_ptr<PIDOutputReceiver> TurnOutput;
+		
         DrivetrainVoltage voltage = DrivetrainVoltage ( 0, 0 );
```
Comment (colescott) [inline src/Subsystems/Drivetrain.h]: DELEEEETEEEEE

**Review by colescott — CHANGES_REQUESTED** [review]
Mostly just remove Voltage stuff from Drvetrain class.

**Comment by colescott** [comment]
Is the NavX being used?

File: src/Subsystems/Drivetrain.h (hunk 5 above)
Comment (colescott) [reply src/Subsystems/Drivetrain.h]: (just the voltage garbage)

**Comment by dengzac** [comment]
NavX is used for turning and staying straight

**Review by colescott — APPROVED** [review]
lgtm

**Pull request merged**